tokio/src/task/error.rs

```diff
@@ -23,7 +23,7 @@ impl JoinError {
     }
 
     /// Create a new `panic` error
-    pub fn panic(err: Box<dyn Any + Send + 'static>) -> JoinError {
+    pub fn panic(err: Box<dyn Any + Send + Sync + 'static>) -> JoinError {
```

Comment:
This is technically a breaking change, right?

Reply:
I *could* put the mutex inside `Repr` and leave the interface unchanged, I guess

Reply:
It could be deprecated and we add panic2?

Reply:
Are public constructors even needed?

Reply:
We could remove for now. I mostly added for completeness

Reply:
I've restricted their to visibility the crate.